MovieBooking/Modules/Authentication/ViewController/SignUpViewController.swift

```diff
@@ -0,0 +1,106 @@
+import UIKit
+import FirebaseAuth
+import FirebaseFirestore
+
+class SignUpViewController: UIViewController, Storyboarded {
+
+    // MARK: - Variables
+    var coordinator: MainCoordinator?
+    var activeTextField: UITextField? = nil
+    var authenticationViewModel = AuthenticationViewModel()
+    
+    // MARK: - Outlets
+    @IBOutlet var viewBackground: UIView!
+    @IBOutlet weak var tfEmail: UITextField!
+    @IBOutlet weak var tfPassword: UITextField!
+    @IBOutlet weak var tfFirstName: UITextField!
+    @IBOutlet weak var tfLastName: UITextField!
+    
+    // MARK: - UIViewController
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        bindViewModel()
+        initialData()
+        // Do any additional setup after loading the view.
+    }
+    
+    // MARK: - File Private Functions
+    fileprivate func initialData() {
+        hideKeyboardWhenTapAround()
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow), name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    fileprivate func bindViewModel() {
+        authenticationViewModel.onSignUpSuccess = { message in
+            print(message)
+            self.generateAlert(validation: message)
+        }
+        authenticationViewModel.signUpValidationError = { message in
+            print(message)
+            self.generateAlert(validation: message)
+        }
+    }
+    
+    fileprivate func generateAlert(validation message: String) {
```

Comment:
if you are using same methods multiple time in project then you can put it in common file so you can reuse it.